pkg/ring/kv/consul/client.go

```diff
@@ -31,6 +32,9 @@ var (
 		MinBackoff: 1 * time.Second,
 		MaxBackoff: 1 * time.Minute,
 	}
+
+	watchKeyRate  = rate.Limit(1)
+	watchKeyBurst = 3
```

Comment:
Can we have these as part of the Config struct below?  And have a way for the default values to be a no-op?  I want this to be opt-in.

Reply:
Moved WatchKeyRate and WatchKeyBurst to Config. Default value to WatchKeyRate (0) disables rate-limiting. Burst is set to 1 if rate-limiting is enabled and Burst is <= 1.